src/features/messages/components/Compose/index.tsx

```diff
@@ -0,0 +1,259 @@
+  native: string
+  short_names: []
+  skin: null
+  unified: string
+}
+
+interface IAudio {
+  blob: Blob
+  startTime: number
+  stopTime: number
+  option: {
+    audioBitsPerSecond: number
+    mimeType: string
+  }
+  blobURL: string
+}
+
+function InputCompose(props: IComposeProps): JSX.Element {
+  const { createMessage, isLoading } = props
+  const [messageTyping, setMessage] = useState('')
+  const [messageImage, setMessageImage] = useState<string[]>([])
+  const [showEmoji, setShowEmoji] = useState(false)
+  const [showRecord, setShowRecord] = useState(false)
+  const [record, setRecord] = useState(false)
+  const [audio, setAudio] = useState('')
+
+  const onSend = () => {
+    createMessage(messageTyping, messageImage, audio)
+    setTimeout(scrollToBottom, 100)
+  }
+
+  const scrollToBottom = () => {
+    const msgList = document.getElementById('wrapper-list-message')
+    if (msgList) {
+      msgList.scrollTo({
+        left: 0,
+        top: msgList.scrollHeight,
+        behavior: 'smooth',
+      })
+    }
+  }
+
+  const handleSendMsg = () => {
+    if (messageTyping || messageImage || audio) {
+      onSend()
+      setMessage('')
+      setMessageImage([])
+      setAudio('')
+      setShowRecord(false)
+    }
+  }
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === 'Enter') {
+      handleSendMsg()
+    }
+  }
+
+  const onChangeInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setMessage(e.target.value)
+  }
+
+  const onChangeEmoji = (emoji: IEmoji) => {
+    setMessage(_message => {
+      return _message + emoji.native
+    })
+  }
+
+  const onChangeShowEmoji = () => {
+    setShowEmoji(!showEmoji)
+  }
+
+  const onChangeShowImage = () => {
+    const element = document.getElementById('input-compose-image')
+    element?.click()
+  }
+
+  const onChangeImageCompose = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const files = e.target.files
+    if (files) {
+      Array.from(files).forEach((file: File) => {
+        const ext = file.name.split('.').pop()?.toLowerCase() || ''
+        if (!['jpg', 'png', 'jpeg'].includes(ext)) {
+          console.log('File invalid')
+        } else {
+          onReadFile(file)
+        }
+      })
+      e.target.value = ''
+    }
+  }
+
+  const onReadFile = (file: File) => {
+    const reader = new FileReader()
+    reader.onload = e => {
+      const { result } = e.target as { result: string }
+      setMessageImage(preImage => [...preImage, result])
+      setTimeout(scrollToBottom, 100)
+    }
+    reader.readAsDataURL(file)
+  }
+
+  const onDeleteImage = (item: string) => () => {
+    const newData = messageImage.filter(key => {
+      return !key.includes(item)
+    })
+    setMessageImage(newData)
+  }
+
+  const onStop = (recordedBlob: IAudio) => {
+    setAudio(recordedBlob.blobURL)
+  }
+
+  const startRecording = () => {
+    setRecord(true)
+  }
+
+  const stopRecording = () => {
+    setRecord(false)
+  }
```

Comment:
Do we really need these functions? They only wrap a single call to the `set...` methods